### PR TITLE
Register metrics API group with NodeMetrics

### PR DESCRIPTION
**Pull request by mksalawa**
Metrics storage logic copied from existing handlers.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/heapster/1304)

### CODE REVIEW

**Comment by piosz** [comment]
@wojtek-t could you please take a look?

**Comment by piosz** [comment]
LGTM from me, @nikhiljindal could you please take a look?

File: metrics/apis/metrics/install/install.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+)
+
+const importPrefix = "k8s.io/heapster/metrics/apis/metrics"
+
+var accessor = meta.NewAccessor()
+
+// availableVersions lists all known external versions for this group from most preferred to least preferred
+var availableVersions = []unversioned.GroupVersion{v1alpha1.SchemeGroupVersion}
+
+func init() {
```
Comment (nikhiljindal) [inline metrics/apis/metrics/install/install.go]: FYI, @deads2k and @lavalamp have changed this install code at many places.
You can look at https://github.com/kubernetes/kubernetes/blob/cf7301f16c036363c4fdcb5d4d0c867720214598/pkg/apis/extensions/install/install.go for an example.

I think @lavalamp wanted to fix it some more before updating all install.go, so this should be fine. But wanted to let you know that we will need to update it soon.

**Comment by nikhiljindal** [comment]
Just one suggestion.
Also, again no tests? Will be good to add some tests :)

File: metrics/apis/metrics/install/install.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
```
Comment (deads2k) [inline metrics/apis/metrics/install/install.go]: Take a look at the other install.go files under `/apis` in authentication, authorization, batch.  They're much smaller.

**Comment by mksalawa** [comment]
@nikhiljindal thanks for the review! The new version looks much better. I would rather update it in the follow-up PR, as it would require godeps update I guess.

As for the tests, we talked about it with @piosz and from what I understand, as the connection to the API server is secured, we need the generated client first to get there in a way kube API is accessed, am I right?

File: metrics/apis/metrics/install/install.go (hunk 2 above)
Comment (piosz) [inline metrics/apis/metrics/install/install.go]: AFAIU your objection is similar to https://github.com/kubernetes/heapster/pull/1304#discussion_r80764878. @mksalawa will fix this in a follow up PR.

**Comment by piosz** [comment]
LGTM,
Thanks @mksalawa for the change!
Thanks @nikhiljindal @deads2k for the review.

**Comment by piosz** [comment]
Travis time-outed.

**Comment by nikhiljindal** [comment]
@mksalawa Yes you will need that for the complete e2e test. But in the meantime you can add unit tests for the code you are adding.
You can also take a look at https://github.com/kubernetes/kubernetes/blob/master/test/integration/examples/apiserver_test.go, which tests an apiserver without enabling auth. It uses simple CURL commands to verify that apiserver comes up fine and expected endpoints exist on the server.

**Pull request merged**